Mergin/version_viewer_dialog.py

```diff
@@ -0,0 +1,639 @@
+# GPLv3 license
+# Copyright Lutra Consulting Limited
+
+from collections import deque
+import os
+import math
+
+from qgis.PyQt import uic, QtCore
+from qgis.PyQt.QtWidgets import (
+    QDialog,
+    QAction,
+    QListWidgetItem,
+    QPushButton,
+    QMenu,
+    QMessageBox,
+    QAbstractItemView,
+    QToolButton,
+)
+from qgis.PyQt.QtGui import QStandardItem, QStandardItemModel, QIcon, QFont, QColor
+from qgis.PyQt.QtCore import (
+    QStringListModel,
+    Qt,
+    QSettings,
+    QModelIndex,
+    QAbstractTableModel,
+    QThread,
+    pyqtSignal,
+    QItemSelectionModel,
+)
+
+from qgis.utils import iface
+from qgis.core import (
+    QgsProject,
+    QgsMessageLog,
+    QgsApplication,
+    QgsFeatureRequest,
+    QgsVectorLayerCache,
+    # Used to filter background map
+    QgsRasterLayer,
+    QgsTiledSceneLayer,
+    QgsVectorTileLayer,
+)
+from qgis.gui import QgsMapToolPan, QgsAttributeTableModel, QgsAttributeTableFilterModel
+
+
+from .utils import (
+    ClientError,
+    icon_path,
+    mergin_project_local_path,
+    PROJS_PER_PAGE,
+    contextual_date,
+    is_versioned_file,
+    icon_path,
+    format_datetime,
+    parse_user_agent,
+    icon_for_layer,
+)
+
+from .mergin.merginproject import MerginProject
+from .mergin.utils import bytes_to_human_size, int_version
+
+from .mergin import MerginClient
+from .diff import make_version_changes_layers
+
+
+ui_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), "ui", "ui_versions_viewer.ui")
+
+
+class VersionsTableModel(QAbstractTableModel):
```

Comment:
Actually it's the other way around, we do use a `QTreeView` but a `QAbstractTableModel` as a model, because right now  the data are flat and `QAbstractTableModel` provides convenient already implemented methods to work with tabular data.

If we want to  expand to tree we would have to change in the feature but have to write more boilre plate code